### ISSUE TITLE
SVG sprite generation

### ISSUE THREAD

**Issue by lumpysimon** [wprig/wprig#36]
I've added a task to the gulpfile to watch for SVG images in mytheme/dev/sprite and create a single SVG sprite from them in mytheme/sprite. The sprite contains symbols with IDs generated from the SVG filenames.

Is this something you would be interested in adding to wprig, or is it outside the scope?

It uses svgmin, svgstore and path:

```
export function sprite() {
	return gulp.src(paths.sprite.src)
	.pipe(svgmin(function(file) {
		var prefix = path.basename(file.relative, path.extname(file.relative));
		return {
			plugins: [{
				cleanupIDs: {
					prefix: prefix + '-',
					minify: true
				}
			}]
		}
	}))
	.pipe(svgstore())
	.pipe(gulp.dest(paths.sprite.dest));
}
```

I'm happy to submit a pull request, but I'm new to Gulp4 so I'm sure my code could be improved (for example, spaces in filenames need to be converted to hyphens in the symbol IDs).

**Comment by mor10** [comment]
This is somewhat related to #35. Let's see if we can add it in when we look at rewiring the Gulp process.

**Comment by lumpysimon** [comment]
Ah yes, this is exactly the sort of thing that a more modular approach to the gulpfile would benefit. I'll keep an eye on #35 and have a go at a pull request for this once I see how the code for that ticket is looking.

**Comment by ataylorme** [comment]
@lumpysimon the gulp improvements are in on the `v1.1` branch so feel free to make a PR against that branch.

Basically, you need to add a new file to the `gulp` directory, perhaps `gulp/svgSprite.js`, with the logic for the task. Then update `gulpfile.babel.js` and `gulp/watch.js` to import the new logic and run it as part of the main task and when SVG files change.

Let me know if you have any questions.

**Comment by lumpysimon** [comment]
Thanks! I'll have a go at that soon

**Comment by ev3rywh3re** [comment]
I have been concentrating on SVG sprite generation at work. Ideally a decent gulp plugin could handle the heavy lifting but I’m still an npm/gulp newbee. 

I know it’s all about final production, so what’s the best sprite implementation with good browser support? ViewBox seems the most compatible across usage methods. It’s such a complex beast I am curious how we could provide a good accessible best practice solution. 

My last stop so far while I’ve been hunting for gulp plugins was: https://css-tricks.com/svg-fragment-identifiers-work/

I really love the ideas of <symbol>, but it doesn’t seem to work well for loading a remote SVG for things like    list-style-image. It still seems that ViewBox has better support. There has to be a nice documentable method here. 

If have any success I will try to share here if I can.

**Comment by ev3rywh3re** [comment]
One more! Should SVG4Everybody be considered for backwards compatibility or is that a decision to be made by us when we use wp-rig? https://github.com/jonathantneal/svg4everybody 

There are so many ideas about how to make things compatible, I wasn't sure if it should even be brought up or if modern current standards and practices are the wp-rig focus.

**Comment by mor10** [comment]
I vote no to svg4everybody. We want to keep this as clean as possible.

**Comment by mor10** [comment]
Punting this to `v2.1` to ensure we get all existing build processes settled before we start adding new features.

**Comment by benoitchantre** [comment]
> Punting this to `v2.1` to ensure we get all existing build processes

@mor10, @ataylorme : is it something we could consider for `v2.0` (SVG sprite generation) or is it tool late?

**Comment by ataylorme** [comment]
We have a feature freeze on `v2.0` and are focusing on bug fixes. Since this is a new feature and not a bug fix, I think it should stay milestoned for a future release.

That being said, after the current bugs in `v2.0` are fixed and we release a beta of `v2.0` we can probably start work on `v2.1`

**Comment by mor10** [comment]
Closing in favor of #454
